Content.Shared/Nutrition/EntitySystems/SatiationSystem.cs

```diff
@@ -0,0 +1,396 @@
+        { SatiationThreashold.Desperate, AlertType.Starving },
+        { SatiationThreashold.Dead, AlertType.Starving }
+    };
+    private AlertCategory HungerAlertCategory = AlertCategory.Hunger;
+
+    private (string, StatusIconPrototype?)[] ThirstIcons = new (string, StatusIconPrototype?)[] {
+        ("ThirstIconOverhydrated", null),
+        ("ThirstIconThirsty", null),
+        ("ThirstIconParched", null)
+    };
+    /// <summary>
+    /// A dictionary relating hunger thresholds to corresponding alerts.
+    /// </summary>
+    private Dictionary<SatiationThreashold, AlertType> ThirstAlertThresholds = new()
+    {
+        { SatiationThreashold.Concerned, AlertType.Thirsty },
+        { SatiationThreashold.Desperate, AlertType.Parched },
+        { SatiationThreashold.Dead, AlertType.Parched }
+    };
+    private AlertCategory ThirstAlertCategory = AlertCategory.Thirst;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        foreach (var pair in ThirstIcons)
+        {
+            var (iconId, prototype) = pair;
+            DebugTools.Assert(_prototype.TryIndex(iconId, out prototype));
+        }
+        foreach (var pair in HungerIcons)
+        {
+            var (iconId, prototype) = pair;
+            DebugTools.Assert(_prototype.TryIndex(iconId, out prototype));
+        }
+
+        SubscribeLocalEvent<SatiationComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<SatiationComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<SatiationComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovespeed);
+        SubscribeLocalEvent<SatiationComponent, RejuvenateEvent>(OnRejuvenate);
+    }
+
+    private void InitializeSatiation(EntityUid uid, Satiation component, Dictionary<SatiationThreashold, AlertType> alertThresholds, AlertCategory alertCategory, MapInitEvent args)
+    {
+        var amount = _random.Next(
+            (int) component.Thresholds[SatiationThreashold.Concerned] + 10,
+            (int) component.Thresholds[SatiationThreashold.Okay]);
+        SetSatiation(component, amount);
+        UpdateCurrentThreshold(component);
+        DoThresholdEffects(uid, component, alertThresholds, alertCategory, false);
+
+        component.CurrentThreshold = GetThreshold(component, component.Current);
+        component.LastThreshold = SatiationThreashold.Okay; // TODO: Potentially change this -> Used Okay because no effects.
+    }
+
+    private void OnMapInit(EntityUid uid, SatiationComponent component, MapInitEvent args)
+    {
+        InitializeSatiation(uid, component.Thirst, ThirstAlertThresholds, ThirstAlertCategory, args);
+        InitializeSatiation(uid, component.Hunger, HungerAlertThresholds, HungerAlertCategory, args);
+        Dirty(uid, component);
+
+        TryComp(uid, out MovementSpeedModifierComponent? moveMod);
+            _movementSpeedModifier.RefreshMovementSpeedModifiers(uid, moveMod);
```

Comment:
Why is second line indented? Is it meant to be an if?

Reply:
Now TryComp is a condition in if statement